packages/services/src/kernel/future.ts

```diff
@@ -221,15 +203,9 @@ export class KernelFutureHandler<
     }
   }
 
-  private async _handleStdin(msg: KernelMessage.IStdinMessage): Promise<void> {
-    let stdin = this._stdin;
-    if (stdin) {
-      // tslint:disable-next-line:await-promise
-      await stdin(msg);
-    }
-  }
-
-  private async _handleIOPub(msg: KernelMessage.IIOPubMessage): Promise<void> {
+  protected async _handleIOPub(
```

Comment:
Same here about moving functions around - probably simpler to just not delete the _handleStdin that was originally there? Unless you want to make the order more consistent, of course.